### ISSUE TITLE
Fix my_current_day record used in a test for an alarm

### ISSUE THREAD

**Issue by elbuenodefali** [zerok/zerokspot.com#160]
Hello,

I was checking your tests here:
https://github.com/zerok/zerokspot.com/blob/486fc52a1ca3a184eedee3b056c4acf8d7a05fc8/content/weblog/2019/testing-prometheus-alerts.md

And I had the same issue. I was able to fix it by making a full series of the record related to the time expression for having its value updated when the alarm expr is evaluated.

I will show you my example:

The alarm: app_jenkins.rules

```
groups:
  - name: Jenkins
    rules:
      - alert: JenkinsAbsentMetrics
        expr: absent_over_time(up{hostname="devel-jenkins-0"}[5m]) and on () business_day
        for: 1m
        labels:
          severity: warning
          slack_channel: "monitoring-jenkins"
        annotations:
          description: "Instance devel-jenkins-0 is not returning metrics for 5m."
          runbook: foo
```
The business_hours.rules

```
groups:
- name: datetime
  rules:
  - record: business_day
    expr: |
       vector(1) and day_of_week() > 0 and day_of_week() < 6 unless count(public_holiday)

  - record: public_holiday
    expr: |
       vector(1) and day_of_month() == 1 and month() == 1
    labels:
       name: newyear

  - record: public_holiday
    expr: |
       vector(1) and day_of_month() == 25 and month() == 12
    labels:
       name: christmas
```

The test for checking the alarm when it is a business_day: app_jenkins_business_day_test.yml

```
rule_files:
  - app_jenkins.rules

evaluation_interval: 1m

# In order to check if it is 'business_day' we need to create the series of 'business_day'
# with 7 samples taken each minute.
# That means, that it will have the value 1 (True) for each minute:
# - series: business_day
#   values: '1+0x7'
# The sample will be '1 1 1 1 1 1 1'
# Therefore, when the alert expression is evaluated, 'business_day' value is True.

tests:
  - interval: 1m
    input_series:
      - series: up{hostname="devel-jenkins-0"}
        values: "0"
      - series: business_day
        values: '1+0x7'

    alert_rule_test:
      - eval_time: 6m
        alertname: JenkinsAbsentMetrics
        exp_alerts: []
      - eval_time: 7m
        alertname: JenkinsAbsentMetrics
        exp_alerts:
          - exp_labels:
              alertname: JenkinsAbsentMetrics
              hostname: devel-jenkins-0
              severity: warning
              slack_channel: monitoring-jenkins
            exp_annotations:
              description: Instance devel-jenkins-0 is not returning metrics for 5m.
              runbook: foo
```
Result:

```
Unit Testing:  app_jenkins_business_day_test.yml
  SUCCESS
```

If I remove this input_series:
```
      - series: business_day
        values: '1+0x7'
```

The test fails the same as you exposed:
```
got:"[]"
```

So, in a nutshell, you need to create the full time-series for your alarm record in order to have the desired value when the `expr` of the alarm is evaluated.

I hope this helps.

Regards

**Comment by zerok** [comment]
@elbuenodefali Thank you 😄 I'll update the post to link to your solution ASAP 🙂
